enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -271,12 +271,17 @@ object SlotAllocation {
         rhsPipeline
 
       case _:CartesianProduct =>
-        val cartesianProductPipeline = lhsPipeline.seedClone()
+        val newPipeline = lhsPipeline.seedClone()
         rhsPipeline.foreachSlot {
           case (k, slot) =>
-            cartesianProductPipeline.add(k, slot)
+            newPipeline.add(k, slot)
         }
-        cartesianProductPipeline
+        newPipeline
+
+      case RollUpApply(_, _, collectionName, _, _) =>
```

Comment:
Could we beef up `SlotAllocationTest` to match? We've been pretty good with the test coverage there, would be nice to keep it up.

Reply:
Yes